packages/legend-query-builder/src/components/watermark/QueryBuilderWatermark.tsx

```diff
@@ -101,20 +109,18 @@ const WatermarkValueEditor = observer(
             dropTargetConnector={dropTargetConnector}
           >
             <BasicValueSpecificationEditor
-              valueSpecification={watermarkValue}
+              valueSpecification={selectedValue}
               setValueSpecification={(val: ValueSpecification): void => {
-                watermarkState.setValue(val);
+                setSelectedValue(deepClone(val));
```

Comment:
is this deepClone necessary, since we already deep clone the orginal value. woudnt this update just the cloned value spec?

Reply:
The `deepClone` seems to be necessary because the `val` parameter in the `setValueSpecification` callback seems to be the same reference as is passed in in the `valueSpecification`. Without the `deepClone` (if we just call `setValueSpecification(val)`) the component doesn't re-render because the prop is technically unchanged (its pointing to the same object). I think if it were a primitive value we wouldn't need the `deepClone`, but since it's an object, the `deepClone` creates a branch new object, which causes the component to re-render since the prop changes.

Reply:
mhm if thats the case, do we need a deepClone for this. just a shallow clone should be enough?

Reply:
Good point. I switched it to shallow clone and it worked fine. So now we only `deepClone` in the `useState` hook to copy the object from the state and shallow clone it when updating it from the editor.